tests/testthat/test-pipgd_dist.R

```diff
@@ -1,3 +1,533 @@
+    
+    names(res_complete$dist_stats) |>
+        expect_equal(c("popshare", "quantile_welfare_share"))
+
+})
+
+
+# Test pipgd_quantile function ####
+# Inputs -------------------------------------------------------------------------
+test_that("pipgd_quantile inputs works as expected", {
+    res1 <- pipgd_quantile(welfare = welfare, weight = weight)
+    res2 <- pipgd_quantile(params = params) 
+
+    # Welfare and weight arguments
+    pipgd_quantile(welfare = welfare, weight = NULL) |>
+        expect_error()
+
+    pipgd_quantile(welfare = NULL, weight = weight) |>
+        expect_error()
+    
+    # Lorenz argument
+    pipgd_quantile(welfare = welfare, weight = weight, lorenz = "Neither NULL, lq or lb") |>
+        expect_error()
+    
+    # n argument 
+    pipgd_quantile(welfare = welfare, weight = weight, n = "Not a valid number") |>
+        expect_error()
+    pipgd_quantile(welfare = welfare, weight = weight, n = -2) |>
+        expect_error()
+    
+    # popshare arguments
+    pipgd_quantile(welfare = welfare, weight = weight, popshare = c(0.3, 0.5, 0.8)) |>
+        expect_no_error()
+    pipgd_quantile(welfare = welfare, weight = weight, popshare = c(0.3, 0.5, -0.8)) |>
+        expect_error()
+
+    # complete argument
+    pipgd_quantile(welfare = welfare, weight = weight, complete = "neither TRUE or FALSE") |>
+        expect_error()
+
+})
+
+# Outputs ----------------------------------------------------------------------------
+test_that("pipgd_quantile outputs work as expected", {
+    
+    # Check same results when params or (welfare and weights) are provided
+    res1 <- pipgd_quantile(welfare = welfare, weight = weight)
+    res2 <- pipgd_quantile(params = params) 
+    expect_equal(res1, res2)
+
+    # Check same results when n or popshare are provided
+    n = 5
+    res_n5 <- pipgd_quantile(welfare = welfare, weight = weight, n = n)
+    res_with_popshare <- pipgd_quantile(welfare = welfare, weight = weight, popshare = c(0.2, 0.4, 0.6, 0.8, 1.0))
+    expect_equal(res_with_popshare, res_n5)
+
+    res_complete <- pipgd_quantile(welfare = welfare, weight = weight, complete = TRUE)
+
+    # Check output type
+    class(res1) |>
+        expect_equal(class(res2))
+    
+    class(res2) |>
+        expect_equal("list")
+    
+    class(res_complete) |>
+        expect_equal("pipgd_params")
+    
+    # Check output length
+    length(res_n5$dist_stats$popshare) |>
+        expect_equal(n)
+
+    length(res_n5$dist_stats$quantile) |>
+        expect_equal(n)
+    
+    # Check popshare output
+    res_n5$dist_stats$popshare[2] |>
+        expect_equal(0.4)
+    
+    # Check quantile output when lorenz = "lb"
+    mean = 1
+    popshare = c(0.2, 0.4, 0.6, 0.8, 1.0)
+    lorenz = "lb"
+
+    res <- pipgd_quantile(welfare = welfare, weight = weight,  popshare = popshare, lorenz = lorenz)
+    qt <- mean*wbpip:::derive_lb(x = popshare, 
```

Comment:
This is a tautology. Please do not compare with `wbpip` functions because the result will always be the same